src/HTTPRequest.cpp

```diff
@@ -48,50 +55,35 @@ void HTTPRequest::download(const URL& url,
     }
 }
 
-void HTTPRequest::post(const URL& url,
-                       const nlohmann::json& data,
-                       std::function<void(const std::string&)> onSuccess,
-                       std::function<void(const std::string&, const long)> onError,
-                       const std::string& fileName,
-                       const std::unordered_set<std::string>& httpHeaders,
-                       const SecureCommunication& secureCommunication,
-                       const std::string& userAgent,
-                       const CurlHandlerTypeEnum& handlerType,
-                       const std::atomic<bool>& shouldRun)
+void HTTPRequest::post(RequestParameters requestParameters,
+                       PostRequestParameters postRequestParameters,
+                       ConfigurationParameters configurationParameters)
 {
-    std::string dataStr;
-    try
-    {
-        dataStr = data.dump();
-    }
-    catch (const std::exception& ex)
-    {
-        onError(ex.what(), NOT_USED);
-        return;
-    }
-
-    post(url, dataStr, std::move(onSuccess), std::move(onError), fileName, httpHeaders, secureCommunication, userAgent);
-}
+    // Request parameters
+    const auto& url {requestParameters.url};
+    const auto& secureCommunication {requestParameters.secureCommunication};
+    const auto& httpHeaders {requestParameters.httpHeaders};
+    // Post request parameters
+    const auto& onError {postRequestParameters.onError};
+    const auto& onSuccess {postRequestParameters.onSuccess};
+    const auto& outputFile {postRequestParameters.outputFile};
+    // Configuration parameters
+    const auto& userAgent {configurationParameters.userAgent};
+    const auto& handlerType {configurationParameters.handlerType};
+    const auto& shouldRun {configurationParameters.shouldRun};
 
-void HTTPRequest::post(const URL& url,
-                       const std::string& data,
-                       std::function<void(const std::string&)> onSuccess,
-                       std::function<void(const std::string&, const long)> onError,
-                       const std::string& fileName,
-                       const std::unordered_set<std::string>& httpHeaders,
-                       const SecureCommunication& secureCommunication,
-                       const std::string& userAgent,
-                       const CurlHandlerTypeEnum& handlerType,
-                       const std::atomic<bool>& shouldRun)
-{
     try
     {
+        std::string data = std::holds_alternative<std::string>(requestParameters.data)
```

Comment:
Improve: Use of const
```suggestion
        const std::string data = std::holds_alternative<std::string>(requestParameters.data)
                                     ? std::get<std::string>(requestParameters.data)
                                     : std::get<nlohmann::json>(requestParameters.data).dump();
```

Reply:
Thank you!
Fixed